lib/open_feature/sdk/configuration.rb

```diff
@@ -14,24 +14,28 @@ class Configuration
       extend Forwardable
 
       attr_accessor :context, :hooks
-      attr_reader :provider
 
-      def_delegator :@provider, :metadata
+      def_delegator :provider, :metadata
 
       def initialize
         @hooks = []
+        @providers = {}
+      end
+
+      def provider(domain: nil)
+        @providers[domain]
       end
 
       # When switching providers, there are a few lifecycle methods that need to be taken care of.
       #   1. If a provider is already set, we need to call `shutdown` on it.
       #   2. On the new provider, call `init`.
       #   3. Finally, set the internal provider to the new provider
-      def provider=(provider)
-        @provider.shutdown if @provider.respond_to?(:shutdown)
+      def set_provider(provider, domain: nil)
+        @providers[domain].shutdown if @providers[domain].respond_to?(:shutdown)
 
         provider.init if provider.respond_to?(:init)
 
-        @provider = provider
+        @providers[domain] = provider
```

Comment:
question: If the domain is nil, we'd be setting the key as nil here, is this what we want?

Reply:
Yep, that was intentional! The default domain would be represented by `nil`.